### ISSUE TITLE
reshape层似乎有bug

### ISSUE THREAD

**Issue by perfyperfect** [Tencent/ncnn#1827]
pytorch转onnx再转ncnn，reshape层似乎有bug，第422层之后，数据清空了
param 如下：
7767517
133 147
Input            data                  0 1 data
Convolution      Conv_0                   1 1 data 205 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=864
BatchNorm        BatchNormalization_1     1 1 205 206 0=32
ReLU             Relu_2                   1 1 206 207
Convolution      Conv_3                   1 1 207 208 0=32 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=9216
BatchNorm        BatchNormalization_4     1 1 208 209 0=32
ReLU             Relu_5                   1 1 209 210
Pooling          MaxPool_6                1 1 210 211 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=0 5=1
Convolution      Conv_7                   1 1 211 212 0=64 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=18432
BatchNorm        BatchNormalization_8     1 1 212 213 0=64
ReLU             Relu_9                   1 1 213 214
Convolution      Conv_10                  1 1 214 215 0=64 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=36864
BatchNorm        BatchNormalization_11    1 1 215 216 0=64
ReLU             Relu_12                  1 1 216 217
Pooling          MaxPool_13               1 1 217 218 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=0 5=1
Convolution      Conv_14                  1 1 218 219 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=8192
BatchNorm        BatchNormalization_15    1 1 219 220 0=128
ReLU             Relu_16                  1 1 220 221
Convolution      Conv_17                  1 1 221 222 0=128 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=147456
BatchNorm        BatchNormalization_18    1 1 222 223 0=128
ReLU             Relu_19                  1 1 223 224
Pooling          MaxPool_20               1 1 224 225 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=1 15=1 5=1
Convolution      Conv_21                  1 1 225 226 0=256 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_22    1 1 226 227 0=256
ReLU             Relu_23                  1 1 227 228
Convolution      Conv_24                  1 1 228 229 0=256 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=589824
BatchNorm        BatchNormalization_25    1 1 229 230 0=256
ReLU             Relu_26                  1 1 230 231
Split            splitncnn_0              1 4 231 231_splitncnn_0 231_splitncnn_1 231_splitncnn_2 231_splitncnn_3
Convolution      Conv_27                  1 1 231_splitncnn_3 232 0=32 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=8192
BatchNorm        BatchNormalization_28    1 1 232 233 0=32
Convolution      Conv_29                  1 1 233 234 0=64 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=18432
BatchNorm        BatchNormalization_30    1 1 234 235 0=64
ReLU             Relu_31                  1 1 235 236
Convolution      Conv_32                  1 1 236 237 0=64 1=3 11=3 2=2 12=2 3=1 13=1 4=2 14=2 15=2 16=2 5=0 6=36864
BatchNorm        BatchNormalization_33    1 1 237 238 0=64
Convolution      Conv_34                  1 1 231_splitncnn_2 239 0=32 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=8192
BatchNorm        BatchNormalization_35    1 1 239 240 0=32
Convolution      Conv_36                  1 1 240 241 0=64 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=18432
BatchNorm        BatchNormalization_37    1 1 241 242 0=64
ReLU             Relu_38                  1 1 242 243
Convolution      Conv_39                  1 1 243 244 0=64 1=3 11=3 2=3 12=3 3=1 13=1 4=3 14=3 15=3 16=3 5=0 6=36864
BatchNorm        BatchNormalization_40    1 1 244 245 0=64
Convolution      Conv_41                  1 1 231_splitncnn_1 246 0=32 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=8192
BatchNorm        BatchNormalization_42    1 1 246 247 0=32
Convolution      Conv_43                  1 1 247 248 0=48 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=13824
BatchNorm        BatchNormalization_44    1 1 248 249 0=48
ReLU             Relu_45                  1 1 249 250
Convolution      Conv_46                  1 1 250 251 0=64 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=27648
BatchNorm        BatchNormalization_47    1 1 251 252 0=64
ReLU             Relu_48                  1 1 252 253
Convolution      Conv_49                  1 1 253 254 0=64 1=3 11=3 2=5 12=5 3=1 13=1 4=5 14=5 15=5 16=5 5=0 6=36864
BatchNorm        BatchNormalization_50    1 1 254 255 0=64
Concat           Concat_51                3 1 238 245 255 256 0=0
Convolution      Conv_52                  1 1 256 257 0=256 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=49152
BatchNorm        BatchNormalization_53    1 1 257 258 0=256
Convolution      Conv_54                  1 1 231_splitncnn_0 259 0=256 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=65536
BatchNorm        BatchNormalization_55    1 1 259 260 0=256
BinaryOp         Add_56                   2 1 258 260 261 0=0
ReLU             Relu_57                  1 1 261 262
Split            splitncnn_1              1 3 262 262_splitncnn_0 262_splitncnn_1 262_splitncnn_2
Pooling          MaxPool_58               1 1 262_splitncnn_2 263 0=0 1=2 11=2 2=2 12=2 3=0 13=0 14=0 15=0 5=1
Convolution      Conv_59                  1 1 263 264 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_60    1 1 264 265 0=128
Convolution      Conv_61                  1 1 265 266 0=256 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=0 6=294912
BatchNorm        BatchNormalization_62    1 1 266 267 0=256
ReLU             Relu_63                  1 1 267 268
Split            splitncnn_2              1 3 268 268_splitncnn_0 268_splitncnn_1 268_splitncnn_2
Convolution      Conv_64                  1 1 268_splitncnn_2 269 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_65    1 1 269 270 0=128
Convolution      Conv_66                  1 1 270 271 0=256 1=3 11=3 2=1 12=1 3=2 13=2 4=1 14=1 15=1 16=1 5=0 6=294912
BatchNorm        BatchNormalization_67    1 1 271 272 0=256
ReLU             Relu_68                  1 1 272 273
Split            splitncnn_3              1 3 273 273_splitncnn_0 273_splitncnn_1 273_splitncnn_2
Convolution      Conv_69                  1 1 273_splitncnn_2 274 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_70    1 1 274 275 0=128
Convolution      Conv_71                  1 1 275 276 0=256 1=3 11=3 2=1 12=1 3=2 13=2 4=1 14=1 15=1 16=1 5=0 6=294912
BatchNorm        BatchNormalization_72    1 1 276 277 0=256
ReLU             Relu_73                  1 1 277 278
Split            splitncnn_4              1 3 278 278_splitncnn_0 278_splitncnn_1 278_splitncnn_2
Convolution      Conv_74                  1 1 278_splitncnn_2 279 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_75    1 1 279 280 0=128
Convolution      Conv_76                  1 1 280 281 0=256 1=3 11=3 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=294912
BatchNorm        BatchNormalization_77    1 1 281 282 0=256
ReLU             Relu_78                  1 1 282 283
Split            splitncnn_5              1 3 283 283_splitncnn_0 283_splitncnn_1 283_splitncnn_2
Convolution      Conv_79                  1 1 283_splitncnn_2 284 0=128 1=1 11=1 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=32768
BatchNorm        BatchNormalization_80    1 1 284 285 0=128
Convolution      Conv_81                  1 1 285 286 0=256 1=3 11=3 2=1 12=1 3=1 13=1 4=0 14=0 15=0 16=0 5=0 6=294912
BatchNorm        BatchNormalization_82    1 1 286 287 0=256
ReLU             Relu_83                  1 1 287 288
Split            splitncnn_6              1 2 288 288_splitncnn_0 288_splitncnn_1
Convolution      Conv_84                  1 1 262_splitncnn_1 289 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=36864
Permute          Transpose_85             1 1 289 290 0=3
Convolution      Conv_86                  1 1 262_splitncnn_0 291 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=18432
Permute          Transpose_87             1 1 291 292 0=3
Convolution      Conv_88                  1 1 268_splitncnn_1 293 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=36864
Permute          Transpose_89             1 1 293 294 0=3
Convolution      Conv_90                  1 1 268_splitncnn_0 295 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=18432
Permute          Transpose_91             1 1 295 296 0=3
Convolution      Conv_92                  1 1 273_splitncnn_1 297 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=36864
Permute          Transpose_93             1 1 297 298 0=3
Convolution      Conv_94                  1 1 273_splitncnn_0 299 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=18432
Permute          Transpose_95             1 1 299 300 0=3
Convolution      Conv_96                  1 1 278_splitncnn_1 301 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=36864
Permute          Transpose_97             1 1 301 302 0=3
Convolution      Conv_98                  1 1 278_splitncnn_0 303 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=18432
Permute          Transpose_99             1 1 303 304 0=3
Convolution      Conv_100                 1 1 283_splitncnn_1 305 0=16 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=36864
Permute          Transpose_101            1 1 305 306 0=3
Convolution      Conv_102                 1 1 283_splitncnn_0 307 0=8 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=18432
Permute          Transpose_103            1 1 307 308 0=3
Convolution      Conv_104                 1 1 288_splitncnn_1 309 0=12 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=27648
Permute          Transpose_105            1 1 309 310 0=3
Convolution      Conv_106                 1 1 288_splitncnn_0 311 0=6 1=3 11=3 2=1 12=1 3=1 13=1 4=1 14=1 15=1 16=1 5=1 6=13824
Permute          Transpose_107            1 1 311 312 0=3
Reshape          Reshape_113              1 1 290 320 0=-1
Reshape          Reshape_119              1 1 294 328 0=-1
Reshape          Reshape_125              1 1 298 336 0=-1
Reshape          Reshape_131              1 1 302 344 0=-1
Reshape          Reshape_137              1 1 306 352 0=-1
Reshape          Reshape_143              1 1 310 360 0=-1
Concat           Concat_144               6 1 320 328 336 344 352 360 361 0=0
Reshape          Reshape_150              1 1 292 369 0=-1
Reshape          Reshape_156              1 1 296 377 0=-1
Reshape          Reshape_162              1 1 300 385 0=-1
Reshape          Reshape_168              1 1 304 393 0=-1
Reshape          Reshape_174              1 1 308 401 0=-1
Reshape          Reshape_180              1 1 312 409 0=-1
Concat           Concat_181               6 1 369 377 385 393 401 409 410 0=0
Reshape          Reshape_187              1 1 361 420 0=4 1=-1
Reshape          Reshape_189              1 1 410 422 0=2
Softmax          Softmax_190              1 1 422 423 0=0 1=1

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
